inference-engine/samples/classification_of_detected_roi/main.cpp

```diff
@@ -0,0 +1,551 @@
+                    } else {
+                        slog::info << clsLabels.at(classes[i]);
+                    }
+                    slog::info << slog::endl;
+                }
+
+                const auto& curImageDims = imageBlob->getTensorDesc().getDims();
+                const auto imageWidth = curImageDims[3];
+                const auto imageHeight = curImageDims[2];
+
+                const auto imageMem = imageBlob->rwmap();
+                const auto imagePtr = imageMem.as<unsigned char*>() +
+                    imageBlob->getTensorDesc().getBlockingDesc().getOffsetPadding();
+
+                addRectangles(imagePtr, imageHeight, imageWidth, boxes, classes, BBOX_THICKNESS);
+
+                const std::string outputFileName = getFileNameFromPath(imageName) + "_out.bmp";
+
+                if (writeOutputBmp(outputFileName, imagePtr, imageHeight, imageWidth)) {
+                    slog::info << "        Stored result to " << outputFileName << slog::endl;
+                } else {
+                    slog::err << "        Failed to write result to " << outputFileName << slog::endl;
+                }
+            };
+
+        const auto getDetectionRectangles =
+            [](const std::vector<Blob::Ptr>& ROIs)
+                    -> std::vector<int> {
+                std::vector<int> boxes;
+                for (Blob::Ptr roiBlob : ROIs) {
+                    const ROI& roi = roiBlob->getROI()->roi;
+                    boxes.push_back(roi.posX);
+                    boxes.push_back(roi.posY);
+                    boxes.push_back(roi.sizeX);
+                    boxes.push_back(roi.sizeY);
+                }
+                return boxes;
+            };
+
+        //
+        // Run detection+classification pipeline
+        //
+
+        slog::info << "Run detection+classification pipeline" << slog::endl;
+
+        const auto createImageBlob =
+            [&imagesData, &imagesWidth, &imagesHeight](size_t imageInd) -> MemoryBlob::Ptr {
+                const auto imageDataPtr = imagesData[imageInd].get();
+
+                const auto imageWidth = imagesWidth[imageInd];
+                const auto imageHeight = imagesHeight[imageInd];
+
+                const auto blobDims = SizeVector {1, 3, imageHeight, imageWidth};
+                const auto blobDesc = TensorDesc(Precision::U8, blobDims, Layout::NHWC);
+
+                return make_shared_blob<uint8_t>(blobDesc, imageDataPtr);
+            };
+
+        for (size_t imageInd = 0; imageInd < images.size(); ++imageInd) {
+            const auto& imageName = images[imageInd];
+            const MemoryBlob::Ptr imageBlob = createImageBlob(imageInd);
+
+            slog::info << "    Process " << imageName << slog::endl;
+
+            detInferRequest.SetBlob(detInputName, imageBlob);
+            detInferRequest.Infer();
+
+            const std::vector<Blob::Ptr> ROIs = getDetectionROIs(as<MemoryBlob>(detInferRequest.GetBlob(detOutputName)), imageBlob);
```

Comment:
@anton-potapov @ilyachur
`getROI()` usage:
Client has the convenience of not dragging detected boxes through the call stack;
Semantics is expressed in a few lines with no mandatory deep-dive;

Compare to cryptic sequence of functions when ROI data is not available for the client: https://github.com/openvinotoolkit/openvino/pull/882/files#diff-b1b9024f2bb9a523502fcb4e2fb3c9d1R505

Reply:
Looks like in alternative solution we should just use vector of blobs in order to save all ROI... It doesn't look like a big impact.

Also I have a question:

Why in this solution should we write more lines of code (551 vs 522)? Both files are new and it is hard to understand what are differen, just code style or alternative solution requires less code lines.

Reply:
>we should just use vector of blobs...

That's the kind of aux data which causes unnecessary inconvenience for the client. This impacts any user code related to multiple ROI blobs.

There are additional handling and informational messages added, after the sample was tested on the image set. They add lines. 
Readability is worth more than amount of lines anyway.

Reply:
From application view, getROI() is helpful, it can simply logical of application to manage ROI list and ROIBlob.
BTW, is it better if put ROI list into ROIblob rather than put one ROI into one ROIBlob?

Reply:
The container of Blobs each having one ROI can be passed through IE API, and then list of ROI for the single inference will be composed by the plugin-specific code.

Reply:
Ok, what does `getROI()` return for usual blob?

Reply:
`nullptr`. 
Two (at least) useful appliances:
* single check whether blob is ROI: `if (blob->getROI()) { ... `
* break the loop over the chain of nested ROI blobs: `for (; blob->getROI(); blob = blob->getROI()->original) `